src/store/modules/DownloadModule.ts

```diff
@@ -121,3 +116,16 @@ export const DownloadModule = {
         }
     },
 }
+
+function indexOfDownloadProgress(allDownloads: DownloadProgress[], assignId: number): number {
+    const index = [...allDownloads].findIndex((downloadProgress: DownloadProgress) => {
```

Comment:
Minor: this too has unnecessarily written out types, as well as unnecessarily using braces with an arrow function.

```typescript
const index = [...allDownloads].findIndex((downloadProgress) => downloadProgress.assignId === assignId);
```